.github/workflows/wheels.yml

```diff
@@ -125,3 +127,44 @@ jobs:
         with:
           name: ${{ matrix.python }}-${{ startsWith(matrix.buildplat[1], 'macosx') && 'macosx' || matrix.buildplat[1] }}
           path: ./wheelhouse/*.whl
+
+      - name: Setup Upload Variables
+        if: ${{ always() }}
+        shell: bash
+        run: |
+          if [ "push" == "${{ github.event_name }}" ] && [ "${{ startsWith(github.ref, 'refs/tags/v') }}" ]; then
+            echo push and tag event
+            echo "ANACONDA_ORG=multibuild-wheels-staging" >> $GITHUB_ENV
+            echo "TOKEN=${{ secrets.NUMPY_STAGING_UPLOAD_TOKEN }}" >> $GITHUB_ENV
+            echo "ANACONDA_UPLOAD=true" >> $GITHUB_ENV
+          elif [ "schedule" == "${{ github.event_name }}" ] || [ "workflow_dispatch" == "${{ github.event_name }}" ]; then
+            echo scheduled or dispatched event
+            echo "ANACONDA_ORG=scipy-wheels-nightly" >> $GITHUB_ENV
+            echo "TOKEN=${{ secrets.NUMPY_NIGHTLY_UPLOAD_TOKEN }}" >> $GITHUB_ENV
+            echo "ANACONDA_UPLOAD=true" >> $GITHUB_ENV
+          else
+            echo non-dispatch event
+            echo "ANACONDA_UPLOAD=false" >> $GITHUB_ENV
+          fi
+      - name: Upload wheels
+        if: ${{ always() }}
```

Comment:
I don't think wheels should be uploaded if any of the build steps above fail.

Reply:
I will check for env.TOKEN.

Reply:
It works, the step was skipped.